File: src/scanner/registry.py
"""
File type registry for scanner: supported extensions and detection.
"""

from pathlib import Path

# Supported extensions (lowercase)
SUPPORTED_EXTENSIONS = {".pdf", ".docx", ".txt", ".xlsx"}


def get_file_type(file_path: str) -> str:
    """Return file type (extension) for supported files, or empty string."""
    ext = Path(file_path).suffix.lower()
    return ext if ext in SUPPORTED_EXTENSIONS else ""


def is_supported(file_path: str) -> bool:
    """Return True if file is supported (PDF, DOCX, TXT, XLSX)."""
    return get_file_type(file_path) != ""


def filter_supported(paths: list[str]) -> list[str]:
    """Filter list of paths to only supported file types."""
    return [p for p in paths if is_supported(p)]

File: src/scanner/registry.cpp
#include "document_ingestion/registry.hpp"
#include <filesystem>
#include <algorithm>

namespace document_ingestion {

static const std::vector<std::string> SUPPORTED_EXTENSIONS = {".pdf", ".docx", ".txt", ".xlsx"};

std::string get_file_type(const std::string& file_path) {
  std::string ext = std::filesystem::path(file_path).extension().string();
  for (auto& c : ext) c = static_cast<char>(std::tolower(static_cast<unsigned char>(c)));
  for (const auto& supported : SUPPORTED_EXTENSIONS) {
    if (ext == supported) return ext;
  }
  return "";
}

bool is_supported(const std::string& file_path) {
  return !get_file_type(file_path).empty();
}

std::vector<std::string> filter_supported(const std::vector<std::string>& paths) {
  std::vector<std::string> result;
  for (const auto& p : paths) {
    if (is_supported(p)) result.push_back(p);
  }
  return result;
}

}  // namespace document_ingestion
